docs/source/advanced/multi_gpu.rst

```diff
@@ -788,6 +788,167 @@ For even more speed benefit, DeepSpeed offers an optimized CPU version of ADAM c
     trainer = Trainer(gpus=4, plugins=DeepSpeedPlugin(cpu_offload=True), precision=16)
     trainer.fit(model)
 
+DeepSpeed ZeRO Stage 3
+""""""""""""""""""""""
+
+DeepSpeed ZeRO Stage 3 shards the optimizer states, gradients and the model parameters (also optionally activations). Sharding model parameters and activations comes with an increase in distributed communication, however allows you to scale your models massively from one GPU to multiple GPUs.
+**The DeepSpeed team report the ability to fine-tune models with over 40B parameters on a single GPU and over 2 Trillion parameters on 512 GPUs.** For more information we suggest checking the `DeepSpeed ZeRO-3 Offload documentation <https://www.deepspeed.ai/news/2021/03/07/zero3-offload.html>`__.
+
+We've ran benchmarks and give a simple example of how all these features in Lightning, which you can see at `minGPT <https://github.com/SeanNaren/minGPT/tree/stage3>`_.
+
+Currently this functionality is only available on master and will be included in our next 1.3 Release Candidate and 1.3 release.
```

Comment:
```suggestion
Currently this functionality is only available on master and will be included in our next 1.3 Release Candidate and 1.3 release.
```
Will this be necessary? As we wouldn't be including this PR in the patch release.

Reply:
Not sure if it is, can remove it people prefer!

Reply:
I would drop it...